packages/tools/examples/contourApi/index.ts

```diff
@@ -0,0 +1,530 @@
+        const viewport = getViewportDisplayingAnnotation(annotation);
+        if (!viewport) {
+          return;
+        }
+        const originalPolyline = convertContourPolylineToCanvasSpace(
+          annotation.data.contour.polyline,
+          viewport
+        );
+
+        console.log(`Annotation ${index + 1}:`);
+        console.log('Original polyline points:', originalPolyline.length);
+
+        try {
+          // Supersample the polyline with target spacing of 1.0
+          const supersampledPolyline = supersamplePolyline2D(
+            originalPolyline,
+            0.1
+          );
+          console.log(
+            'Supersampled polyline points:',
+            supersampledPolyline.length
+          );
+
+          // Update the annotation with the supersampled polyline
+          annotation.data.contour.polyline = convertContourPolylineToWorld(
+            supersampledPolyline,
+            viewport
+          );
+        } catch (error) {
+          console.error('Error supersampling polyline:', error);
+        }
+      }
+    });
+
+    // Re-render to show the updated polylines
+    renderingEngine.renderViewports(viewportIds);
+  },
+});
+
+addButtonToToolbar({
+  title: 'Remove Small Islands',
+  onClick: () => {
+    const annotations = cornerstoneTools.annotation.state.getAllAnnotations();
+
+    if (annotations.length === 0) {
+      console.log('No annotations found. Please draw some contours first.');
+      return;
+    }
+
+    // Collect all polylines
+    const polylines = [];
+    const annotationMap = new Map();
+
+    annotations.forEach((annotation) => {
+      if (annotation?.data?.contour?.polyline) {
+        const polyline = getAnnotationPolyline(annotation);
+        polylines.push(polyline);
+        annotationMap.set(polyline, annotation);
+      }
+    });
+
+    if (polylines.length === 0) {
+      console.log('No valid polylines found.');
+      return;
+    }
+
+    console.log('Original polylines count:', polylines.length);
+
+    try {
+      // Remove islands smaller than area threshold of 300
+      const islandPolylines = findIslands(polylines, 300);
+      console.log('Island polylines count:', islandPolylines.length);
+
+      // Remove annotations that were filtered out
+      islandPolylines.forEach((islandIndex) => {
+        const polyline = polylines[islandIndex];
+        const annotation = annotationMap.get(polyline);
+        if (annotation) {
+          csToolsAnnotation.state.removeAnnotation(annotation.annotationUID);
+        }
+      });
+
+      // Re-render to show the updated annotations
+      const renderingEngine = getRenderingEngine(renderingEngineId);
+      renderingEngine.renderViewports(viewportIds);
+    } catch (error) {
+      console.error('Error removing islands:', error);
+    }
+  },
+});
+
+addButtonToToolbar({
+  title: 'Decimate Polylines',
+  onClick: () => {
+    const annotations = cornerstoneTools.annotation.state.getAllAnnotations();
```

Comment:
cornerstoneTools.segmentation.utilities.decimateContours({ segmentationId, segmentIndex options: { epsilon })

should use the logic in `packages/tools/src/utilities/math/polyline/decimate.ts`